js/src/utils/toScopeState.test.js

```diff
@@ -0,0 +1,156 @@
+/**
+ * Internal dependencies
+ */
+import toScopeState from './toScopeState';
+
+const SCOPE = {
+	// Manage product listings and accounts for Google Shopping
+	CONTENT: 'https://www.googleapis.com/auth/content',
+	// Manage new site verifications with Google
+	SITE_VERIFICATION_VERIFY_ONLY:
+		'https://www.googleapis.com/auth/siteverification.verify_only',
+	// Manage AdWords campaigns
+	AD_WORDS: 'https://www.googleapis.com/auth/adwords',
+};
```

Comment:
Sorry, I don't get your point. In current ES practice, even if an object or array is declared with `const`, devs usually don't mutate its properties, and almost never saw a dev mutate the properties of an all caps variable name.

If we were developing a library and had to expose constants to third parties for importing, or if we were considering some particular performance tuning, then I think it's a good practice to apply `Object.freeze`. Otherwise, in an app development context, I think it should be ok to just declare a constant and protect it by common conventions. But I may miss something, do you have any concerns?

Reply:
You have a point above. I'm not particularly concerned about `Object.freeze` above, so we can put that aside.

My main concern here is that since now there are two same  `SCOPE` object in `toScopeState.js` and `toScopeState.test.js`, do we want to consider refactor them out?

(The `Object.freeze` suggestion is a secondary concern to the above main concern; if we are not going to refactor them out, then we don't need to use `Object.freeze`.)

Reply:
I looked deeper into the code and figured that the `scopes` parameter in `toScopeState` accepts an array of strings, so the `SCOPE` object in `toScopeState.test.js` is more like a convenience object for our testing here, i.e. we can choose to not have duplicated `SCOPE` object and instead use string array like `['https://www.googleapis.com/auth/content']`, and the tests would still work.

In other words, I think we don't need to refactor this. 😆

Reply:
The `SCOPE` in *toScopeState.js* file are the expected values. And the `SCOPE` in *test* file are used to simulate the `scopes` array returned from Google API. I think they should not be the same source, otherwise, if the `SCOPE` values in *toScopeState.js* file are changed accidentally, the tests could still pass.

I adjusted it by https://github.com/woocommerce/google-listings-and-ads/pull/1071/commits/1c88e84056e8668ef124ecc24b0f80cd4ee821ba to make them look different and added a comment to avoid confusion.

Reply:
> The `SCOPE` in _toScopeState.js_ file are the expected values. And the `SCOPE` in _test_ file are used to simulate the `scopes` array returned from Google API. I think they should not be the same source, otherwise, if the `SCOPE` values in _toScopeState.js_ file are changed accidentally, the tests could still pass.

Yeah, agree 👍  My mind was playing tricks on me when I saw the same `SCOPE` objects. Sorry about that. 😅